source/Halibut.Tests/Diagnostics/ExceptionReturnedByHalibutProxyExtensionMethodFixture.cs

```diff
@@ -0,0 +1,211 @@
+                    tentacleListening.Trust(Certificates.OctopusPublicThumbprint);
+
+                    var echo = octopus.CreateClient<IEchoService>("https://localhost:" + tentaclePort, "Wrong Thumbrprint");
+
+                    Assert.Throws<HalibutClientException>(() => echo.SayHello("Hello"))
+                        .IsNetworkError()
+                        .Should()
+                        .Be(HalibutNetworkExceptionType.NotANetworkError);
+                }
+            }
+            
+            [Test]
+            public void BecauseTheDataStreamHadAnErrorOpeningTheFileWithFileStream_WhenSendingToListening_ItIsNotANetworkError()
+            {
+                var services = new DelegateServiceFactory();
+                services.Register<IEchoService>(() => new EchoService());
+                using (var octopus = new HalibutRuntime(Certificates.Octopus))
+                using (var tentacleListening = new HalibutRuntime(services, Certificates.TentacleListening))
+                {
+                    var tentaclePort = tentacleListening.Listen();
+                    tentacleListening.Trust(Certificates.OctopusPublicThumbprint);
+
+                    var echo = octopus.CreateClient<IEchoService>("https://localhost:" + tentaclePort, Certificates.TentacleListeningPublicThumbprint);
+
+                    var dataStream = new DataStream(10, stream =>
+                    {
+                        new FileStream("DoesNotExist2497546", FileMode.Open).Dispose();
+                    });
+                    Assert.Throws<HalibutClientException>(() => echo.CountBytes(dataStream))
+                        .IsNetworkError()
+                        .Should()
+                        .Be(HalibutNetworkExceptionType.NotANetworkError);
+                }
+            }
+            
+            [Test]
+            public void BecauseTheDataStreamHadAnErrorOpeningTheFileWithFileStream_WhenSendingToPolling_ItIsNotANetworkError()
+            {
+                var services = new DelegateServiceFactory();
+                services.Register<IEchoService>(() => new EchoService());
+                using (var octopus = new HalibutRuntime(Certificates.Octopus))
+                using (var tentaclePolling = new HalibutRuntime(services, Certificates.TentaclePolling))
+                {
+                    var octopusPort = octopus.Listen();
+                    octopus.Trust(Certificates.TentaclePollingPublicThumbprint);
+
+                    tentaclePolling.Poll(new Uri("poll://SQ-TENTAPOLL"), new ServiceEndPoint(new Uri("https://localhost:" + octopusPort), Certificates.OctopusPublicThumbprint));
+
+                    var echo = octopus.CreateClient<IEchoService>("poll://SQ-TENTAPOLL", Certificates.TentaclePollingPublicThumbprint);
+
+                    var dataStream = new DataStream(10, stream =>
+                    {
+                        new FileStream("DoesNotExist2497546", FileMode.Open).Dispose();
```

Comment:
The error message from FileStream opening a file which is what Server is doing looks different to the one thrown by just throwing a FileNotFoundException. This and the above tests exists because we want to be sure we get this one correct.